src/app/shared/components/breadcrumb.tsx

```diff
@@ -3,165 +3,127 @@
-                    key: 'Devices',
-                    text: t(ResourceKeys.breadcrumb.devices)
-                });
-                return;
-            }
-
-            if (pathComponent.toLowerCase() === ROUTE_PARTS.DEVICE_DETAIL.toLowerCase()) {
-                const deviceId = this.getDeviceIdFromSearch(search);
-                items.push({
-                    href: `#/${ROUTE_PARTS.RESOURCE}/${hubName}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.DEVICE_DETAIL}/${ROUTE_PARTS.IDENTITY}?${ROUTE_PARAMS.DEVICE_ID}=${deviceId}`,
-                    key: 'Device',
-                    text: deviceId
-                });
-
-                if (index < pathComponents.length - 1 && pathComponents[index + 1].toLowerCase() !== ROUTE_PARTS.DIGITAL_TWINS.toLowerCase()) {
-                    const remainingPathComponents = pathComponents.slice(index + 1);
-                    items.push(...this.buildDeviceDetails(remainingPathComponents, hubName, deviceId, t));
-                }
-                return;
-            }
-
-            if (pathComponent.toLowerCase() === ROUTE_PARTS.DIGITAL_TWINS.toLowerCase()) {
-                const deviceId = this.getDeviceIdFromSearch(search);
-                const interfaceId = this.getInterfaceIdFromSearch(search);
-                items.push({
-                    href: `#/${ROUTE_PARTS.RESOURCE}/${hubName}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.DEVICE_DETAIL}/${ROUTE_PARTS.DIGITAL_TWINS}/${ROUTE_PARTS.INTERFACES}/?${ROUTE_PARAMS.DEVICE_ID}=${deviceId}&${ROUTE_PARAMS.INTERFACE_ID}=${interfaceId}`,
-                    key: `device_${deviceId}_${interfaceId}`,
-                    text: interfaceId
-                });
-
-                if (index < pathComponents.length - 1) {
-                    const remainingPathComponents = pathComponents.slice(index + 1);
-                    items.push(...this.buildDigitalTwinDetails(remainingPathComponents, hubName, deviceId, interfaceId, t));
-                }
-                return;
+    // tslint:disable-next-line:cyclomatic-complexity
+    private renderBreadcrumbItem = (t: TranslationFunction, route: string | undefined) => {
+        if (route) {
+            switch (route) {
+                case ROUTE_PARTS.RESOURCE:
+                    return <></>;
+                case this.props.hostName:
+                    return this.renderTextItem(t(ResourceKeys.breadcrumb.hub, {hubName: this.getShortHubName(route)}));
+                case ROUTE_PARTS.DEVICE_DETAIL:
+                    return this.renderTextItem(getDeviceIdFromQueryString(this.props));
+                case ROUTE_PARTS.DIGITAL_TWINS:
+                    return this.renderTextItem(getInterfaceIdFromQueryString(this.props));
+                case ROUTE_PARTS.MODULE_IDENTITY:
+                    return this.renderLinkItem(this.getLocalizedKey(t, route), [ROUTE_PARAMS.DEVICE_ID]);
+                default:
+                    return this.renderLinkItem(this.getLocalizedKey(t, route));
```

Comment:
you are unnecessarily passing this 't' around all these functions.. this is a stateless component ..should be pretty easy to use localization hook.